### PR TITLE
chore(docs): fix docs for static assets (`public`)

### PR DESCRIPTION
**Pull request by styfle**
Closes NEXT-2260

### CODE REVIEW

File: docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx [hunk 1]
```diff
@@ -8,22 +8,26 @@ description: Next.js allows you to serve static files, like images, in the publi
 
 Next.js can serve static files, like images, under a folder called `public` in the root directory. Files inside `public` can then be referenced by your code starting from the base URL (`/`).
 
-For example, if you add `me.png` inside `public`, the following code will access the image:
+For example, the file `public/avatars/me.png` can be viewed by visiting the `/avatars/me.png` path. The code might look like:
 
-```jsx filename="Avatar.js"
+```jsx filename="Example.js"
```
Comment (leerob) [inline docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx:13]:
```suggestion
```jsx filename="avatar.js"
```

File: docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx [hunk 2]
```diff
@@ -8,22 +8,26 @@ description: Next.js allows you to serve static files, like images, in the publi
 
 Next.js can serve static files, like images, under a folder called `public` in the root directory. Files inside `public` can then be referenced by your code starting from the base URL (`/`).
 
-For example, if you add `me.png` inside `public`, the following code will access the image:
+For example, the file `public/avatars/me.png` can be viewed by visiting the `/avatars/me.png` path. The code might look like:
```
Comment (leerob) [inline docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx:11]:
```suggestion
For example, the file `public/avatars/me.png` can be viewed by visiting the `/avatars/me.png` path. The code to display that image might look like:
```

File: docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx [hunk 3]
```diff
@@ -8,22 +8,26 @@ description: Next.js allows you to serve static files, like images, in the publi
...
+```jsx filename="Example.js"
 import Image from 'next/image'
 
-export function Avatar() {
-  return <Image src="/me.png" alt="me" width="64" height="64" />
+export function Avatar({ id, alt }) {
+  return <Image src={`/avatars/${id}.png`} alt={alt} width="64" height="64" />
+}
+
+export function AvatarOfMe() {
```
Comment (leerob) [inline docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx:20]: Why have two separate components here?

**Review by Ethan-Arrowood — APPROVED** [review]

File: docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx (hunk 3 above)
Comment (styfle) [reply docs/02-app/01-building-your-application/06-optimizing/11-static-assets.mdx:20]: I'm demonstrating a dynamic `src` string here.

The previous example wasn't clear why you would use `<Image src="/me.jpg" />` vs `import me from './me.jpg; <Image src={me} />` so the new example makes it clear that it works with dynamic url string.

**Pull request merged**